src/pip/_internal/req/constructors.py

```diff
@@ -334,6 +334,11 @@ def install_req_from_req_string(
             "%s depends on %s " % (comes_from.name, req)
         )
 
+    if req.url:
```

Comment:
Would it be helpful to add a code comment explaining why this if block is needed (e.g. what this PR addresses)?